.github/workflows/run-kola-tests.yaml

```diff
@@ -111,11 +238,24 @@ jobs:
           # run the test.
           test_run ${{ matrix.arch }} qemu_uefi
 
+          test_run ${{ matrix.arch }} qemu_update
+
+          # Stop the background webserver
+          set +e
+          kill %1
+          set -e
+
       - name: Upload artifacts
         if: always()
         uses: actions/upload-artifact@v3
         with:
-          name: test-results-${{ matrix.arch }}
+          name: ${{ matrix.arch }}-test-results
           path: |
-            scripts/__TESTS__
-            scripts/results-.*.tap
+            scripts/__TESTS__/*/_kola_temp/
+            scripts/results-*.tap
+
+      - name: Create Test Summary
+        if: always()
+        uses: test-summary/action@v2
+        with:
+          paths: "scripts/results-*.tap"
```

Comment:
These files need to have the first line one of the following patters:

https://github.com/test-summary/action/blob/b266d9a22ea23552e00a4ee4276543880f5f686e/src/test_parser.ts#L299-L303

Otherwise it tries to parse them as XML, and you get the ```Error: Non-whitespace before first tag.``` message.

Reply:
It appears that we need to prepend a magic "TAP version 13" line. There are a [few examples in their unit tests](https://github.com/test-summary/action/tree/b266d9a22ea23552e00a4ee4276543880f5f686e/test/resources/tap) that suggest the format we emit should otherwise work. I'll patch and re-run.

Reply:
Oh wow that TAP parser is brittle. It insists on 2 spaces and who knows what else for embedded YAML. This makes me sad. TAP is a loosely defined format and it is meant to be interpreted loosely, not strictly.

I'll step away from that action and will investigate other options.